# Train - Test

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
from sklearn.model_selection import train_test_split

df = pd.read_csv('test_estandarizado.csv')

y = df['ParentSupport']
X = df.drop(columns = ['ParentSupport'])

# División 80% entrenamiento, 20% prueba
X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.2, random_state=42)
X_train, X_val, Y_train, Y_val = train_test_split(X_train, Y_train, test_size=0.2, random_state=42)

print("Tamaño de X_train:", len(X_train))
print("Tamaño de X_val:", len(X_val))
print("Tamaño de X_test:", len(X_test))

Tamaño de X_train: 493039
Tamaño de X_val: 123260
Tamaño de X_test: 154075


In [ ]:
X_train.to_csv("X_train.csv", index=False)
Y_train.to_csv("Y_train.csv", index=False)
X_val.to_csv("X_val.csv", index=False)
Y_val.to_csv("Y_val.csv", index=False)
X_test.to_csv("X_test.csv", index=False)
Y_test.to_csv("Y_test.csv", index=False)

In [ ]:
from sklearn.linear_model import LogisticRegression
clf = LogisticRegression(random_state=0, max_iter=1000).fit(X_train, Y_train)
clf.predict(X.iloc[:2, :])
clf.predict_proba(X.iloc[:2, :])
clf.score(X_test, Y_test)

0.6329320136297257

In [ ]:
# --- FORMA CORRECTA ---

# 1. Crear el modelo
clf = LogisticRegression(random_state=0, max_iter=1000)

# 2. Entrenar el modelo SOLAMENTE con los datos de entrenamiento
clf.fit(X_train, Y_train)

# 3. Evaluar el modelo SOLAMENTE con los datos de prueba (datos que NUNCA ha visto)
score_de_prueba = clf.score(X_test, Y_test)



**Este resultado significa la precisión del modelo, es decir, el 63% de las veces el modelo podrá predecir acertadamente la variable "ParentSupport" habiendo sido entrenado con todos los datos**